articles/terraform/provider-version-history-azurerm-3-0-0-to-3-116-0.md

```diff
@@ -333,7 +333,7 @@ ENHANCEMENTS:
 * `network/virtualnetworks` - update to use `hashicorp/go-azure-sdk` ([#26217](https://github.com/hashicorp/terraform-provider-azurerm/issues/26217))
 * `network/virtualwans`: update route resources to use `hashicorp/go-azure-sdk` ([#26189](https://github.com/hashicorp/terraform-provider-azurerm/issues/26189))
 * `azurerm_container_app_job` - support for the `key_vault_secret_id` and `identity`  properties in the  `secret` block ([#25969](https://github.com/hashicorp/terraform-provider-azurerm/issues/25969))
-* `azurerm_kubernetes_cluster` -  support forthe  `dns_zone_ids` popperty in the `web_app_routing` block ([#26117](https://github.com/hashicorp/terraform-provider-azurerm/issues/26117))
+* `azurerm_kubernetes_cluster` -  support for the  `dns_zone_ids` property in the `web_app_routing` block ([#26117](https://github.com/hashicorp/terraform-provider-azurerm/issues/26117))
```

Comment:
```suggestion
* `azurerm_kubernetes_cluster` - support for the `dns_zone_ids` property in the `web_app_routing` block ([#26117](https://github.com/hashicorp/terraform-provider-azurerm/issues/26117))
```

Reply:
@KarlErickson There are more places with duplicate spaces, i didnt look for those as i consider those grammar and not spelling. There are going to be more grammar fixing, after i finish overall about 60 PRs for all the content ;)

Reply:
This is amazing stuff. I'm not sure how other author teams / other repo owners will respond or what special requests they may have, but when working with the azure-dev-docs repo, is there any way you could create one pull request per article? ( assume you're automating this in some way.) The reason: since different authors own different articles, that degree of atomicity would make it easier to accept the changes (and moreover, just grok what belongs to me versus someone else on my team). If not, no worries. Great work.

Reply:
@ArieHein duplicate spaces is not really an issue, as it doesn't change the rendering. I just tend to clean that up when I see it. :-)

Reply:
As im doing this to multiple docs that MS runs and to various other tools, i can tell you that originally for the MS azure i have over 700 spell fixes only. Im trying to break to 10-20 per PR or per sections to make it easy to review, but thats a lot of work creating 50 branches and then creating PRs for each while other people also contribute across the repo.
(which potentially means merge conflicts that i happily fix on my side).

To tackle grammar and even Markdown rules failures is even more work going as i think the last time i checked that went up to almost 3000 files across the entire azure docs, thats 3000 PRs if we go per doc, and at that level is not as much of a hobby but actual job :) 

What i can recommend is that you take example from @sdwheeler that does a great work on the PowerShell docs for when it comes to markdown and even spelling, with using dictionaries , as i think by now the v7.5 is as clean of any spell issues that I have started creating PRs for grammar and markdown fixes to align with the contribution style guide.

I try to follow the same contribution guide that Sean did in the PowerShell docs and i hope there will be some alignment across MS products. It should not be different authors have different styles.
What should be the case is tooling that runs both on dev pc/codespace, rules in the repo for mandatory extensions and having a policy in the PR process both in your internal repos and public facing ones.
Needless to say, all docs added / changed after the merge of fixes will deviate at some point if these steps i mentioned will not be created.

just trying to raise awareness of the importance of correct documentation, and its not just for MS products :)

Reply:
> @ArieHein duplicate spaces is not really an issue, as it doesn't change the rendering. I just tend to clean that up when I see it. :-)

technically its wasted bytes in size of file, multiply by potential versions, that in the end aggregates to MB and thats ofc all in tcpip packets over the internet :)
Its like the story about the kids who persuaded some office to change the fonts used and save money on ink and printer lifespan. Consider spaces the binary equivalent ;)

Reply:
Hi @ArieHein - I really appreciate your volunteer work here, but I recommend that you focus on fixing only obvious, straightforward errors, such as spelling mistakes and formatting problems that cause incorrect rendering. More substantial edits, such as grammar fixes, would take much longer to review, and might cause subtle changes of meaning that would need review and approval from subject matter experts. These delays could then also result in merge conflicts as the affected files are updated by other pull requests.

Thanks also for bringing these issues to our attention. I agree that it's important to improve our processes as much as possible to prevent errors, and we'll definitely look into this.